src/util/opengltexture2d.cpp

```diff
@@ -20,7 +20,7 @@ void OpenGLTexture2D::setData(const QPixmap& pixmap) {
     setData(pixmap.toImage());
 };
 
-void OpenGLTexture2D::setData(const QSharedPointer<Paintable>& pPaintable) {
+void OpenGLTexture2D::setData(const std::shared_ptr<Paintable>& pPaintable) {
     if (pPaintable) {
         setData(pPaintable->toImage());
     }
```

Comment:
this is not a suggestion for this PR, just a coding guideline in the future. If you don't need to transfer ownership, just take a plain ptr (if you need nullability) or even just a ref to the underlying type directly and leave the rest to the caller. Then this method works regardless how the pointer/type is managed. Does that make sense?
```suggestion
void OpenGLTexture2D::setData(const Paintable* pPaintable) {
    if (pPaintable) {
        setData(pPaintable->toImage());
    }
```

Reply:
In the pointer type it results in the caller doing a semi-pretty `unique_ptr::get()`, but in the reference case, its really ergonomic since you just need to deref, the ptr.